kamon-core/src/main/scala/kamon/trace/TracingContext.scala

```diff
@@ -66,7 +66,7 @@ private[trace] class TracingContext(traceName: String, token: String, izOpen: Bo
       val segment = currentSegments.next()
       if (segment.isClosed)
         segmentsInfo += segment.createSegmentInfo(_startTimestamp, startTimestamp)
-      else
+      else if (log != null)
```

Comment:
has this ever been null in the tests that you have performed? I',m not sure of why the check is being made.